core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -62,7 +62,12 @@ object AclCommand {
     var authorizerProperties = Map.empty[String, Any]
     if (opts.options.has(opts.authorizerPropertiesOpt)) {
       val props = opts.options.valuesOf(opts.authorizerPropertiesOpt).asScala.map(_.split("="))
-      props.foreach(pair => authorizerProperties += (pair(0).trim -> pair(1).trim))
+      props.foreach {
+        case Array(key: String, value: String) => authorizerProperties += (key.trim -> value.trim)
+        case array =>
+          println(s"Invalid authorizer property ${array.mkString(",")}")
```

Comment:
> Would it make sense to just throw an exception here and let the catch in main handle and exit?

Sure.

> An alternate option is using CommandLineUtils.parseKeyValueArgs for this functionality. It looks like the ConsoleConsumer and ConsoleProducer use that for their --property arguments.

`CommandLineUtils.parseKeyValueArgs` will not detect a malformed property. As it just creates a key-value pair, with empty value for keys with missing values. I feel that might let unintended errors pass, e.g., `--authorizer-properties zookeeper.connect localhost:2181` will create two properties `zookeeper.connect` and `localhost:2181` with empty values.